lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -478,7 +480,7 @@ def headers(options = {})
           "X-Stripe-Client-User-Agent" => user_agent,
           "X-Stripe-Client-User-Metadata" => {:ip => options[:ip]}.to_json
         }
-        headers.merge!("Idempotency-Key" => idempotency_key) if idempotency_key
+        headers.merge!("Idempotency-Key" => idempotency_key) if idempotency_key && @enable_idempotency
```

Comment:
Yeah, I get that. I'd usually shy away from mutating input as well, but in this case, I think it's better than having potentially leaky state.